joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstForFunctionsCreator.scala

```diff
@@ -101,7 +101,9 @@ trait AstForFunctionsCreator(implicit withSchemaValidation: ValidationMode) { th
       val baseStmtBlockAst = astForMethodBody(node.body, optionalStatementList)
       transformAsClosureBody(refs, baseStmtBlockAst)
     } else {
-      if (methodName != Defines.Initialize && methodName != Defines.InitializeClass) {
+      if (methodName == Defines.TypeDeclBody) {
+        astForMethodBody(node.body, optionalStatementList, returnLastExpression = false)
```

Comment:
Instead of introducing the boolean and having to make special accommodations in `astForMethodBody`, you could perhaps just confine the edge case to this
```suggestion
        astForStatementList(StatementList(optionalStatementList.statements ++ stmtList.statements)(stmtList.span))
```